src/sentry/discover/endpoints/discover_saved_queries.py

```diff
@@ -47,26 +47,25 @@ def get(self, request, organization):
         sort_by = request.query_params.get("sortBy")
         if sort_by in ("name", "-name"):
             order_by = sort_by
-            paginator_cls = OffsetPaginator
         elif sort_by in ("dateCreated", "-dateCreated"):
             order_by = "-date_created" if sort_by.startswith("-") else "date_created"
-            paginator_cls = DateTimePaginator
         elif sort_by in ("dateUpdated", "-dateUpdated"):
             order_by = "-date_updated" if sort_by.startswith("-") else "date_updated"
-            paginator_cls = DateTimePaginator
         else:
             order_by = "name"
-            paginator_cls = OffsetPaginator
+        queryset = queryset.order_by(order_by)
 
         # Old discover expects all queries and uses this parameter.
         if request.query_params.get("all") == "1":
             saved_queries = list(queryset.all())
             return Response(serialize(saved_queries), status=200)
+
+        def data_fn(offset, limit):
+            return list(queryset[offset : offset + limit])
```

Comment:
```suggestion
            return list(queryset[offset : offset + limit + 1])
```

@markstory  Otherwise we only retrieve items from `offset` to `offset + limit - 1`.

Reply:
The generic offset paginator needs one extra row to detect if there are more results on the next page though.

Reply:
Oops looks like I got the math wrong 👍 

```
end = offset + limit - 1
end - offset + 1 === limit
```